client/ayon_houdini/plugins/create/create_usd_componentbuilder.py

```diff
@@ -0,0 +1,70 @@
+import inspect
+
+from ayon_houdini.api import plugin
+from ayon_core.pipeline import CreatedInstance, CreatorError
+
+import hou
+
+
+class CreateUSDComponentBuilder(plugin.HoudiniCreator):
+    identifier = "io.ayon.creators.houdini.componentbuilder"
+    label = "USD Component Builder LOPs"
+    product_type = "usd"
+    icon = "cubes"
+    description = "Create USD from Component Builder LOPs"
+
+    def get_detail_description(self):
+        return inspect.cleandoc("""
+            Creates a USD publish from a Component Output LOP that is part of 
+            a solaris component builder network.
+            
+            The created USD will contain the component builder LOPs and all its
+            dependencies inside the single product.
+            
+            To use it, select a Component Output LOP and click "Create" for
+            this creator. It will generate an instance for each selected
+            Component Output LOP.
+        """)
+
+    def create(self, product_name, instance_data, pre_create_data):
+        nodes = hou.selectedNodes()
+        builders = [
+            node for node in nodes 
+            if node.type().nameWithCategory() == "Lop/componentoutput"
+        ]
+        for builder in builders:
+            self.create_for_instance_node(product_name, instance_data, builder)
+
+    def create_for_instance_node(
+            self, product_name, instance_data, instance_node):
+
+        try:
+            self.customize_node_look(instance_node)
+            instance_data["instance_node"] = instance_node.path()
+            instance_data["instance_id"] = instance_node.path()
+            instance_data["families"] = self.get_publish_families()
+            instance = CreatedInstance(
+                self.product_type,
+                product_name,
+                instance_data,
+                self)
+            self._add_instance_to_context(instance)
+            self.imprint(instance_node, instance.data_to_store())
+        except hou.Error as er:
+            raise CreatorError("Creator error: {}".format(er)) from er
+
```

Comment:
This is essential for contribution workflow to work properly.
```suggestion

        instance_node.setParms({
            # Set the 'default prim' by default to the folder name being
            # published to
            "rootprim": '/`strsplit(chs("folderPath"), "/", -1)`',
        })
```

Reply:
Let's not do this. The component builder is more likely to be used without contribution workflow.

Reply:
What about considering the component builder `Validate USD Asset Contribution Default Prim` so that users can get a hint about wha is going wrong? maybe we can add an issue for it for reference?

Reply:
> I believe the validator you describe is already in place?

Yes, I was mainly talking about [ValidateUSDAssetContributionDefaultPrim](https://github.com/ynput/ayon-houdini/blob/develop/client/ayon_houdini/plugins/publish/validate_usd_asset_contribution_default_prim.py)

So, I did a small experiment to reuse it with component builder.
Here's it

https://github.com/BigRoy/ayon-houdini/compare/enhancement/147-ay-7047_publishing-component-builder-usd-while-maintaining-hierarchical-structure...ynput:ayon-houdini:enhancement/147-ay-7047_publishing-component-builder-usd-while-maintaining-hierarchical-structure-add-rootprim-validation

---
I've found a potential bug here, if the default prim was modified manually and the user forgot to add `/` at the beginning. this case will pass the following condition 

https://github.com/ynput/ayon-houdini/blob/fb03bd0b91bbf7c9ef701b940eae40884ec56c2a/client/ayon_houdini/plugins/publish/validate_usd_asset_contribution_default_prim.py#L63-L63

Reply:
I see. I think it does look fine the way you implemented it - I just would recommend not doing it at all, because it's not a production use case at all as far as I know. It could be worth leaving this code somewhere memorable, e.g. a draft PR that you close so that it's easier found back later if we do want to implement it for new reasons.